### ISSUE TITLE
java language stack overflow

### ISSUE THREAD

**Issue by W6BJK** [arduino/Arduino#2091]
I get the message:
Exception in thread "Thread-583" java.lang.StackOverflowError
followed by a long list of other messages while trying to
compile an Arduino sketch.  The problem appears to be
my attempt to define a 4 by n table.

```C++
const byte Set2a[][4] = {
  {0   ,0   ,0   ,UNUSED},        // 00 unused
.......
  {0xF7,0   ,0   ,FUNKEY}         // 83 Function key F7
}; 
```

 Attempting to interface a PS/2 keyboard to an adduino.

**Comment by ffissore** [comment]
Please give a try to the IDEs linked at the bottom of this email on the dev list
https://groups.google.com/a/arduino.cc/forum/#!msg/developers/4X2T3rCxXWM/YNJl6PZuDucJ
We're testing a new preprocessor and it compiles your sketch just fine.

**Comment by ffissore** [comment]
New preprocessor tracked at #2636. Builds for testing it are available

**Comment by ffissore** [comment]
Fixed by #3779
